esrally/reporter.py

```diff
@@ -945,15 +1018,16 @@ def _safe_divide(n, d):
         if as_percentage:
             diff = _safe_divide(contender - baseline, baseline) * 100.0
             precision = 2
-            suffix = "%"
+            formatted = f"{diff:.{precision}f}%"
         else:
-            diff = formatter(contender - baseline)
+            diff = contender - baseline
             precision = 5
-            suffix = ""
+            formatted = formatter(diff)
+            if not isinstance(formatted, str):
+                formatted = f"{diff:.{precision}f}"
 
         # ensures that numbers that appear as "zero" are also colored neutrally
         threshold = 10 ** -precision
-        formatted = f"{diff:.{precision}f}{suffix}"
```

Comment:
This is suspicious yes, where do you spit out strings?

Reply:
This change appears to cause two issues. First, we ignore precision in some cases:

```
|                                   90th percentile service time |   range_field_conjunction_small_range_big_term_query |    187.54434117581695 |     ms |
|                                   99th percentile service time |   range_field_conjunction_small_range_big_term_query |    194.86027733539228 |     ms |
|                                  100th percentile service time |   range_field_conjunction_small_range_big_term_query |    216.29926899913698 |     ms |
```

Second, the unit of disk usage stats is in "Value" instead of being in "Unit":

```|                                                         Metric |                                                 Task |                 Value |   Unit |
|---------------------------------------------------------------:|-----------------------------------------------------:|----------------------:|-------:|
|                          weather-data-2016 _primary_term total |                                                      |               0 bytes |        |
|                               weather-data-2016 _version total |                                                      |               0 bytes |        |
|                          weather-data-2016 WT17 inverted index |                                                      |              15 bytes |        |
|                              weather-data-2016 WT17 doc values |                                                      |             110 bytes |        |
|                                   weather-data-2016 WT17 total |                                                      |             125 bytes |        |
```

Reply:
> I saw the precision changes but couldn't figure out how they came about. I'll dig into it this morning.

Yesterday I *thought* I hadn't changed the precision code. I also thought that the precision only changed when I turned on the telemetry device. It turns out that this this commit changes the precision whether or not the telemetry device is enabled. *and* enabling the telemetry device further changes the precision. Fun.

Disk usage stats on:
```
|                     Cumulative indexing time of primary shards |                                                      |  0.0018833333333333334 |    min |
```

Disk usage stats off:
```
|                     Cumulative indexing time of primary shards |                                                      |  0.00211667 |    min |
```

Master:
```
|                     Cumulative indexing time of primary shards |                                                      |      0.0017 |    min |
```